library/core/src/num/f16.rs

```diff
@@ -744,16 +744,24 @@ impl f16 {
     #[inline]
     #[unstable(feature = "f16", issue = "116909")]
     // #[unstable(feature = "float_minimum_maximum", issue = "91079")]
+    #[rustc_const_unstable(feature = "f16", issue = "116909")]
     #[must_use = "this returns the result of the comparison, without modifying either input"]
     pub const fn maximum(self, other: f16) -> f16 {
-        if self > other {
-            self
-        } else if other > self {
-            other
-        } else if self == other {
-            if self.is_sign_positive() && other.is_sign_negative() { self } else { other }
-        } else {
-            self + other
+        #[cfg(not(bootstrap))]
+        {
+            intrinsics::maximumf16(self, other)
+        }
+        #[cfg(bootstrap)]
+        {
+            if self > other {
+                self
+            } else if other > self {
+                other
+            } else if self == other {
+                if self.is_sign_positive() && other.is_sign_negative() { self } else { other }
+            } else {
+                self + other
+            }
```

Comment:
Would it be worth preserving this as "fallback" implemention (by giving the intrinsic a body)? Then const-eval wouldn't need an implementation at all, and also cranelift and GCC would not need any extra work.

Reply:
We could, but it's not currently done `min` and `max`, @tgross35 was that intentional?

Reply:
`min` and `max` are *way* older than our ability to have fallback bodies.

Reply:
Sure, but as part of adding `f16` and `f128` we could have added a fallback, but it wasn't done. Hence my question if it was intentional that fallback bodies haven't been added.

Reply:
> Also the minnum16 and minnum128 intrinsics are quite recent

No, definitely not.^^
https://doc.rust-lang.org/1.50.0/std/?search=minnumf32

Reply:
I have added a fallback for `{minimum,maximum}{16,128}`, GCC and Cranelift should have full support.

I didn't touched const-eval as we have the necessary support in `rustc_apfloat`.

Reply:
> We could, but it's not currently done `min` and `max`, @tgross35 was that intentional?

To follow up, when I did the original implementation I tried to keep things as similar as possible to the existing float types. Maybe it makes sense to update those later?